services/familyClassifiers.service.ts

```diff
@@ -0,0 +1,180 @@
+'use strict';
+
+import moleculer, {Context} from 'moleculer';
+import { Method, Service } from 'moleculer-decorators';
+
+import DbConnection from '../mixins/database.mixin';
+import {
+  COMMON_DEFAULT_SCOPES,
+  COMMON_FIELDS,
+  COMMON_SCOPES,
+  CommonFields,
+  CommonPopulates, FieldHookCallback,
+  RestrictionType,
+  Table,
+} from '../types';
+import {User} from "./users.service";
+
+
+interface Fields extends CommonFields {
+  id: number;
+  name: string;
+  nameLatin: string;
+
+}
+
+interface Populates extends CommonPopulates {}
+
+export type FamilyClassifier<
+  P extends keyof Populates = never,
+  F extends keyof (Fields & Populates) = keyof Fields,
+> = Table<Fields, Populates, P, F>;
+
+@Service({
+  name: 'familyClassifiers',
+  mixins: [DbConnection({
+    collection: 'familyClassifiers',
+  })],
+  settings: {
+    fields: {
+      id: {
+        type: 'string',
+        columnType: 'integer',
+        primaryKey: true,
+        secure: true,
+      },
+      name: 'string|required',
+      nameLatin: {
+        type: 'string',
+        required: true,
+        validate: 'validateLatinName'
+      },
+      species: {
+        type: 'array',
+        virtual: true,
+        readonly: true,
+        default: (): any[] => [],
+        async populate(ctx: Context, _values: any, families: User[]) {
+          return await Promise.all(
+              families.map(async (family) =>
+                  ctx.call('speciesClassifiers.find', {
+                    query: {
+                      family: family.id,
+                    },
+                  }),
+              ),
+          );
+        },
+      },
```

Comment:
tokie populate'ai yra "skaudus" ir leti (gali nulausti DB). Kodel?
- kuomet kvieciama `service.find` -> sukuriamas 1 DB connection
- kuomet is praeito `find` kiekvienam instance'ui kvieciamas po 1 `service2.find` gaunasi tiek DB papildomu connection'u kiek 1-ame yra instance'u.

Pvz: Jei `service.find` grazina 100 instance'u, tai bendrai turesime 101 DB connection'a (tolygiai auga su instance'u kieki). Jei pool'as yra nusettintas 7-iem connection'am, tai 1) trukme tokio istraukimo labai isauga 2) kiti request'ai esantys is kitu naudotoju - stovi eileje

Sprendimas: https://github.com/AplinkosMinisterija/biip-shared-configuration/blob/main/packages/moleculer-accounts/src/utils/index.ts#L17-L89 siulau suiinstaliuoti si package'a (jei reik pagalbos - padeciau) ir tada naudoti `PopulateHandlerFn` helperi, kuris minetuoju atveju sukurs tik 2 connectionus - 1 root'ui ir 1 vaikams